File: app/src/main/java/com/example/gemini_ai_chat/GeminiModels.java

package com.example.gemini_ai_chat;

import java.util.Collections;
import java.util.List;

public class GeminiModels {

    public static class Request {
        public List<Content> contents;

        public Request(String text) {
            this.contents = Collections.singletonList(new Content(Collections.singletonList(new Part(text))));
        }
    }

    public static class Content {
        public List<Part> parts;

        public Content(List<Part> parts) {
            this.parts = parts;
        }
    }

    public static class Part {
        public String text;

        public Part(String text) {
            this.text = text;
        }
    }

    public static class Response {
        public List<Candidate> candidates;
    }

    public static class Candidate {
        public Content content;
    }
}

File: app/src/main/java/com/example/gemini_ai_chat/GeminiModels.kt

package com.example.gemini_ai_chat

object GeminiModels {
    // भेजने के लिए (Request)
    data class Request(val contents: List<Content>)

    data class Content(val parts: List<Part>)

    data class Part(val text: String)

    // जवाब पाने के लिए (Response)
    data class Response(val candidates: List<Candidate>)

    data class Candidate(val content: Content)
}
